File: v-chat/src/main/java/me/linx/vchat/netty/handler/ConnectHandler.java
package me.linx.vchat.netty.handler;

import io.netty.channel.ChannelHandlerAdapter;
import io.netty.channel.ChannelHandlerContext;
import me.linx.vchat.netty.NettyServerListener;

public class ConnectHandler extends ChannelHandlerAdapter {

    @Override
    public void channelActive(ChannelHandlerContext ctx) throws Exception {
        super.channelActive(ctx);
        NettyServerListener.clientSum.increment();
    }

    @Override
    public void channelInactive(ChannelHandlerContext ctx) throws Exception {
        super.channelInactive(ctx);
        NettyServerListener.clientSum.decrement();
    }

    @Override
    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
        super.exceptionCaught(ctx, cause);
    }

}

File: v-chat-app/app/src/main/java/me/linx/vchat/app/data/im/handler/ConnectHandler.kt
package me.linx.vchat.app.data.im.handler

import io.netty.channel.ChannelHandlerAdapter
import io.netty.channel.ChannelHandlerContext

class ConnectHandler : ChannelHandlerAdapter() {

    override fun exceptionCaught(ctx: ChannelHandlerContext?, cause: Throwable?) {
        cause?.printStackTrace()
        ctx?.channel()?.close()
    }
}
